### PR TITLE
Refactored query string building to account for array handling

### PR DESCRIPTION
**Pull request by Jan0707**
### WHY are these changes introduced?

Duplicates https://github.com/Shopify/shopify-api-php/pull/232, as described there:
> The function \Shopify\Utils::validateHmac() does not work when query parameters contain array values, this is because http_build_query will transform array values to a different format than Shopify uses when generating the hash, see https://community.shopify.com/c/shopify-apis-and-sdks/hmac-calculation-vs-ids-arrays/m-p/261154.

### WHAT is this pull request doing?

This PR outsources the generation of the query string to a dedicated function that takes care of all the Shopify-specifics for this case: namely ensuring we do not include the HMAC in the string and use the custom format for arrays.

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above
- [x] I have added/updated tests for this change
- [ ] I have updated the documentation for public APIs from the library (if applicable)

### CODE REVIEW

File: src/Utils.php [hunk 1]
```diff
@@ -60,6 +60,42 @@ public static function sanitizeShopDomain(string $shop, ?string $myshopifyDomain
         }
     }
 
+    /**
+     * Builds query strings that are compatible with Shopify's format for array handling
+     * Example: IDs = [1,2,3]
+     * PHP would generate:  ids[]=1&IDs[]=2&IDs[]=3
+     * Shopify expects:     ids=["1","2","3"] (URL encoded)
```
Comment (paulomarg) [inline src/Utils.php:67]: What would this look like for associative arrays?

**Review by paulomarg — APPROVED** [review]
LGTM, just one non-blocking question!

File: src/Utils.php (hunk 1 above)
Comment (Jan0707) [reply src/Utils.php:67]: I was wondering the same thing! I even thought about various ways of implementing those, but

1) I could not find a standards/RFC document that clearly outlined how that was _supposed_ to look - only found a couple of resources demonstrating how some languages/frameworks handle that.
2) I was not sure whether we need this.
3) I was not sure what our server side implementation of that looked like.

TL;DR: Too much uncertainty and no proven use-case at hand. Was not enough to keep me from shipping this in time for our next release.
